### PR TITLE
TEE_GetNextPersistentObject: objectInfo is optional

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

Fix https://github.com/OP-TEE/optee_os/issues/757

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
I would have preferred dealing with this directly in `TEE_GetNextPersistentObject()` instead, but this works also of course.

The commit message is a bit misleading as we're actually not even touching that function in the commit.

**Comment by pascal-brand-st-dev** [comment]
Right. And then it is easier to follow the GP spec. I'll update.

**Comment by pascal-brand-st-dev** [comment]
updated

File: lib/libutee/tee_api_objects.c [hunk 1]
```diff
@@ -653,9 +655,13 @@ TEE_Result TEE_GetNextPersistentObject(TEE_ObjectEnumHandle objectEnumerator,
...
 	}
 
+	if (objectInfo)
+		pt_info = objectInfo;
+	else
+		pt_info = &local_info;
 	len = *objectIDLen;
 	res = utee_storage_next_enum((unsigned long)objectEnumerator,
-				     objectInfo, objectID, &len);
+				     pt_info, objectID, &len);
```
Comment (jforissier) [inline lib/libutee/tee_api_objects.c]: I would have used `objectInfo ? objectInfo : &local_info` here but that's fine.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**